src/utils/index.ts

```diff
@@ -0,0 +1,157 @@
+    if (isNullish(value) || typeof value !== 'object') {
+        return false
+    }
+
+    const proto = Object.getPrototypeOf(value)
+    return isNullish(proto) || proto === Object.prototype
+}
+
+export const template = (tpl: string, options: InterpolateOptions = {}) => {
+    // Matches either:
+    //   - `${key}` with key captured in group 1,
+    //   - `{{<key>}}` with key captured in group 2, or
+    //   - `<key>` with key captured in group 3.
+    const defaultPattern = /(?:\${([\s\S]+?)}|{{<([\s\S]+?)>}}|<([\s\S]+?)>)/g
+    const pattern = options.interpolate || defaultPattern
+
+    return (data: Record<string, unknown>) =>
+        tpl.replace(pattern, (_match, g1, g2, g3): string => {
+            const key = (g1 || g2 || g3 || '').trim()
+            return key in data ? String(data[key]) : ''
+        })
+}
+
+export const mapValues = <T, U>(obj: Record<string, T>, fn: (value: T) => U) =>
+    Object.entries(obj).reduce<Record<string, U>>((acc, [key, value]) => {
+        acc[key] = fn(value)
+        return acc
+    }, {})
+
+export const findKey = <T>(obj: Record<string, T>, predicate: Predicate<T>): string | undefined =>
+    Object.keys(obj).find((key) => predicate(obj[key]))
+
+export const pick = <T extends PlainObject, K extends keyof T>(
+    obj: T | null | undefined,
+    keys: K[]
+): Partial<T> => {
+    if (isNullish(obj)) return {}
+    return keys.reduce<Partial<T>>((acc, key) => {
+        if (Object.prototype.hasOwnProperty.call(obj, key)) {
+            acc[key] = obj[key]
+        }
+        return acc
+    }, {})
+}
+
+export const omit = <T extends PlainObject, K extends keyof T>(
+    obj: T | null | undefined,
+    keys: K[]
+): Partial<T> => {
+    if (obj === undefined || obj === null) return {}
+
+    return Object.keys(obj)
+        .filter((key) => !keys.includes(key as K))
+        .reduce<Partial<T>>((acc, key) => {
+            acc[key as K] = obj[key as K]
+            return acc
+        }, {})
+}
+
+export const partial =
+    // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+        <T extends (...args: any[]) => any>(fn: T, ...partials: any[]) =>
+        // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+        (...args: any[]) =>
+            fn(...partials, ...args)
+
+export const getError = (error: unknown): VeggiesError => {
+    if (isObject(error)) return { ...error, message: error['message'] as string }
+    if (typeof error === 'string') return { message: error }
+    if (!error) return { message: 'unknown error' }
+    return { message: JSON.stringify(error) }
+}
+
+export const getType = (value: unknown): string => {
+    if (Array.isArray(value)) return 'array'
+    if (value === null) return 'null'
+    if (value instanceof Date) return 'date'
+    if (value instanceof RegExp) return 'regexp'
+    if (typeof value === 'object' && value !== null && 'nodeType' in value && value.nodeType === 1)
```

Comment:
```suggestion
    if (typeof value === 'object' && 'nodeType' in value && value.nodeType === 1) return 'element'
```

Reply:
Yes, the condition `'nodeType' in value` is already bypass the "null" (object). Thank!

Reply:
Thanks